text/0034-precomputed-entity-attributes.md

```diff
@@ -0,0 +1,186 @@
+(probably via a wrapper around the Core implementation, as we do for many other
+Core types).
+Note that today, `EvalResult` plays the role of `Value` in our public API, but
+conversion between `EvalResult` and `Value` is relatively expensive (`O(n)`).
+We could either:
+
+- expose `Value` but leave `EvalResult` alone. This provides maximum backwards
+compatibility, but might cause confusion, as `Value` and `EvalResult`
+would be conceptually similar, but slightly different and not interchangeable.
+- consolidate `EvalResult` and `Value` into a single representation, called
+`Value` and based on the Core `Value`. This provides maximum API cleanliness
+but requires the most breaking changes.
+- expose `Value`, but keep `EvalResult` as a type synonym for `Value`. This
+is almost the same as the above option, but keeps the `EvalResult` type name
+around, even though it makes breaking changes to the structure of
+`EvalResult`. This might reduce confusion for users who are migrating, as
+they only have to adapt to the changes to `EvalResult` and not rename
+`EvalResult` to `Value` in their own code; but, it might increase confusion
+for future users, who will see two names for the same thing.
+- not expose `Value` and just use `EvalResult`, even in the new constructors.
+This provides maximum backwards compatibility, like the first option, and
+also avoids the possible confusion generated by the first option, but comes
+at a notable performance cost (due to the inefficient
+`EvalResult`-to-`Value` conversion required under the hood) and might
+generate a different kind of confusion as `EvalResult` is an awkward name in
+the context of these constructors.
+
+This RFC currently proposes we take the third option.
+
+### Component 3: Remove no-longer-necessary interface to precompute `Value`s
+
+Third and finally, we remove `Entities::evaluate()`, as all `Entities` are now
+stored in their evaluated form by default.
+This addresses the second point in [Motivation](#motivation).
+
+This is not a breaking change because `Entities::evaluate()` has not yet been
+released in any Cedar release.
+
+## Drawbacks
+
+- This RFC represents some breaking changes for users upgrading to a new major
+version of Cedar. All breaking changes come with some cost in terms of user
+experience for existing users. This RFC contends that the benefit outweighs the
+cost in this case.
+
+## Alternatives
+
+### Alternative A
+
+We could do only first and third components of this RFC, and not the second
+component, at least for now.
+(The second component -- adding the new constructors for `Entities` and `Entity`
+-- is severable and could be its own RFC, or perhaps a change that doesn't rise
+to the level of requiring an RFC.)
+
+In essence, we could decide that the second component is more trouble than it's
+worth -- that it provides only marginal benefit to users at the cost of
+cluttering our API, both with additional constructors, and with the public
+`Value` type.
```

Comment:
I tend to agree with this assessment.

Reply:
Updated this RFC to move the second component to Alternative A (and adopt the previous Alternative A as the main proposal).  See my comment below.